jgrapht-core/src/main/java/org/jgrapht/alg/decomposition/HeavyPathDecomposition.java

```diff
@@ -0,0 +1,538 @@
+ * of v, including v itself. We define a tree edge (v,parent(v)) to be heavy if $2*size(v)$ &gt; $size(parent(v))$
+ * and light, otherwise.
+ *
+ * The set of heavy edges form the edges of the decomposition.
+ *
+ * <p>
+ *  A benefit of this decomposition is that on any root-to-leaf path of a tree with n nodes,
+ *  there can be at most $log_2(n)$ light edges.
+ *
+ * <p>
+ *   This implementation runs in $O(|V|)$ time and requires $O(|V|)$ extra memory, where $|V|$ is the number of
+ *   vertices in the tree/forest.
+ *
+ * <p>
+ *   Note: If an edge is not reachable from any of the roots provided, then that edge is neither light
+ *   nor heavy.
+ * </p>
+ *
+ * @author Alexandru Valeanu
+ * @since June 2018
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class HeavyPathDecomposition<V, E> implements TreeToPathDecompositionAlgorithm<V, E> {
+
+    private final Graph<V, E> graph;
+    private final Set<V> roots;
+
+    private Map<V, Integer> vertexMap;
+    private List<V> indexList;
+
+    private int[] sizeSubtree, parent, depth, component;
+    private int[] path, lengthPath, positionInPath, firstNodeInPath;
+
+    private int numberOfPaths;
+    private List<List<V>> paths;
+
+    private Set<E> heavyEdges;
+    private Set<E> lightEdges;
+
+    /**
+     * Create an instance with a reference to the tree that we will decompose and to the root of the tree.
+     *
+     * Note: The constructor will NOT check if the input tree is a valid tree.
+     *
+     * @param tree the input tree
+     * @param root the root of the tree
+     */
+    public HeavyPathDecomposition(Graph<V, E> tree, V root) {
+        this(tree, Collections.singleton(Objects.requireNonNull(root, "root cannot be null")));
+    }
+
+    /**
+     * Create an instance with a reference to the forest that we will decompose and to the sets of roots of the
+     * forest (one root per tree).
+     *
+     * Note: If two roots appear in the same tree, an error will be thrown.
+     * Note: The constructor will NOT check if the input forest is a valid forest.
+     *
+     * @param forest the input forest
+     * @param roots the set of roots of the graph
+     */
+    public HeavyPathDecomposition(Graph<V, E> forest, Set<V> roots) {
+        this.graph = Objects.requireNonNull(forest, "input tree/forrest cannot be null");
+        this.roots = Objects.requireNonNull(roots, "set of roots cannot be null");
+
+        decompose();
+    }
+
+    private void allocateArrays(){
+        final int n = graph.vertexSet().size();
+
+        sizeSubtree = new int[n];
+        parent = new int[n];
+        depth = new int[n];
+        component = new int[n];
+
+        path = new int[n];
+        lengthPath = new int[n];
+        positionInPath = new int[n];
+
+        heavyEdges = new HashSet<>();
+        lightEdges = new HashSet<>();
+    }
+
+    private void normalizeGraph(){
+        /**
```

Comment:
There is some kind of communication barrier here :)  This comment was previously in the desired format:

        /*
         * Normalize the graph: map each vertex to an integer (using a HashMap) keep the reverse
         * mapping (using an ArrayList)
         */

That's the way all of them should be if you are going with slash star format.

Reply:
I finally understood what you meant. I decided to go with the classical //